crates/autopilot/src/limit_order_quoter.rs

```diff
@@ -60,63 +66,12 @@ impl LimitOrderQuoter {
                 failed_orders += join_all(chunk.iter().map(|order| async move {
-                    match self
-                        .quoter
-                        .calculate_quote(QuoteParameters {
-                            sell_token: order.data.sell_token,
-                            buy_token: order.data.buy_token,
-                            side: match order.data.kind {
-                                OrderKind::Buy => OrderQuoteSide::Buy {
-                                    buy_amount_after_fee: order.data.buy_amount,
-                                },
-                                OrderKind::Sell => OrderQuoteSide::Sell {
-                                    sell_amount: SellAmount::BeforeFee {
-                                        value: order.data.sell_amount + order.data.fee_amount,
-                                    },
-                                },
-                            },
-                            from: order.metadata.owner,
-                            app_data: order.data.app_data,
-                            signing_scheme: match order.signature {
-                                Signature::Eip712(_) => QuoteSigningScheme::Eip712,
-                                Signature::EthSign(_) => QuoteSigningScheme::EthSign,
-                                Signature::Eip1271(_) => QuoteSigningScheme::Eip1271 {
-                                    onchain_order: false,
-                                    verification_gas_limit: default_verification_gas_limit(),
-                                },
-                                Signature::PreSign => QuoteSigningScheme::PreSign {
-                                    onchain_order: false,
-                                },
-                            },
-                        })
-                        .await
-                    {
-                        Ok(quote) => {
-                            if let Err(err) = self
-                                .database
-                                .update_limit_order_fees(
-                                    &order.metadata.uid,
-                                    &FeeUpdate {
-                                        surplus_fee: quote.fee_amount,
-                                        full_fee_amount: quote.full_fee_amount,
-                                    },
-                                )
-                                .await
-                            {
-                                tracing::error!(
-                                    ?err,
-                                    ?quote,
-                                    "failed to update quote surplus fee, skipping"
-                                );
-                                true
-                            } else {
-                                false
-                            }
-                        }
+                    match self.update_surplus_fee(order).await {
```

Comment:
In case you want the error counting a little tighter you could do 👇 . I guess it's a matter of personal taste if you consider this an improvement. No strong opinion here.
```rust
                failed_orders += join_all(chunk.iter().map(|order| async move {
                    let update_result = self.update_surplus_fee(order).await;
                    if let Err(err) = &update_result {
                        let order_uid = &order.metadata.uid;
                        tracing::warn!(%order_uid, ?err, "skipped limit order due to error");
                    };
                    usize::from(update_result.is_err())
                }))
                .await
                .into_iter()
                .sum::<usize>();
```